imod/util/structured.py

```diff
@@ -98,7 +98,7 @@ def _replace(
         out[replaceable] = value[indices[replaceable]]
         return out.reshape(a.shape)
 
-    if is_scalar(to_replace):
+    if isinstance(to_replace, int | float):
```

Comment:
``is_scalar`` does more than only testing if ``int`` or ``float``. For example a np.array(2.0) is also possible or xr.DataArray(3). What error does it print? Otherwise just ignore it with a docstring as done on other places.

Reply:
the issue here is in fact line 104. On that line we use` np.isnan(to_replace)`, and `np.isnan` does not necessarily work on all the types that `to_replace ` could have. So in my initial solution I fixed this by limiting the options to int or float, for which `np.isnan` defenitely works. 
Now I reverted to leaving line 101 untouched (so using` is_scalar`)  and I added the `type:ignore` to line 104